### ISSUE TITLE
Assignment in conditional must be allowed in while()

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#1316]
I think that it could be a new option (enabled by default, maybe). 

![image](https://user-images.githubusercontent.com/421851/57117744-e0dfc880-6d34-11e9-8cc3-9407c8be0578.png)

I prefer allow it in this case than do that (or using `do while()`):

![image](https://user-images.githubusercontent.com/421851/57117792-1d132900-6d35-11e9-90a8-07e9412226a4.png)

**Comment by funivan** [comment]
Maybe you can use `explode` ;)
In our codebase, we often write the following code:
```php
while($page = $grabber->getNextPage()){ ... }
```
And at first, I think that option for this inspection would be a good thing. But then I realize that we can use `iterator` and skip using `while`
```php
foreach(new Pages($grabber) as $page){ ... }
```
So i`m not against this option but in most cases we can rewrite our code.

**Comment by rentalhost** [comment]
Really, maybe is a good time to review the code and improve it, but in this case, for instance, changing for something like you said will create a needless overhead because it is a very simple looping that breaks the string into something like that:

```php
the_process('it.is.just.an.example') === [
  'it.is.just.an.example',
  'it.is.just.an',
  'it.is.just',
  'it.is',
  'it',
];
```

So I can check if a part of string exists on this array (it need _starts with_ and be exploded by dot each single part). So the `explode()` itself will not do the job alone. :(

**Comment by kalessil** [comment]
Checked, seems to be a native IDE inspection.

**Comment by rentalhost** [comment]
Oh, you are right! :D I do not have checked correctly. I will report it to Jetbrains.
